packages/theming/src/utils/StyledBaseIcon.ts

```diff
@@ -5,12 +5,13 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
-import styled from 'styled-components';
+import styled, { ExecutionProps } from 'styled-components';
 import React, { Children } from 'react';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-export const StyledBaseIcon = styled(({ children, theme, ...props }) =>
-  React.cloneElement(Children.only(children), props)
+export const StyledBaseIcon = styled(
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  ({ children, theme, ...props }: React.HTMLProps<any> & ExecutionProps) =>
+    React.cloneElement(Children.only(children as React.ReactElement), props)
```

Comment:
```suggestion
import { Children, cloneElement, PropsWithChildren, ReactElement, SVGAttributes } from 'react';
import styled, { DefaultTheme } from 'styled-components';

interface IStyledBaseIconProps extends PropsWithChildren<SVGAttributes<SVGElement>> {
  theme: DefaultTheme;
}

export const StyledBaseIcon = styled(
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  ({ children, theme, ...props }: IStyledBaseIconProps) =>
    cloneElement(Children.only(children as ReactElement), props)
)`
  /* empty-source */
`;
```

This provides a tighter type definition. As a result, you're going to get a couple more `tsc` errors that need attention. The interface should be exported via `index.ts`.